platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/DelegatingPersistenceManager.java

```diff
@@ -65,25 +69,37 @@ public void close() throws Exception {
 
   @Override
   public void store(String pid, Dictionary properties) throws IOException {
-    ConfigurationContextImpl context = createContext(pid, properties);
-    if (context.shouldBeVisibleToPlugins()) {
-      List<ConfigurationPersistencePlugin> plugins =
-          asPlugins(configPersistenceTracker.getServices());
-      for (ConfigurationPersistencePlugin plugin : plugins) {
-        plugin.handleStore(context);
+    // Read lock because Felix provides its own protection for multiple config deletes
```

Comment:
❓ Do we need that comment? Or are we locking just to make sure that an initialization is not currently happening? Same in the other methods